app_feup/lib/controller/networking/network_router.dart

```diff
@@ -177,34 +178,65 @@ class NetworkRouter {
 
   static Future<List<Trip>> getNextArrivalsStop(
       String stopCode, BusStopData stopData) async {
-    final String url =
-        'http://move-me.mobi/NextArrivals/GetScheds?providerName=STCP&stopCode=STCP_' +
-            stopCode;
+
+    final url =
+        'https://www.stcp.pt/pt/itinerarium/soapclient.php?codigo=' + stopCode;
+
     final http.Response response = await http.get(url);
-    final List<Trip> tripList = List();
+    final htmlResponse = parse(response.body);
 
-    final List json = jsonDecode(response.body);
+    final tableEntries =
+        htmlResponse.querySelectorAll('#smsBusResults > tbody > tr.even');
 
-    for (var TripKey in json) {
-      final trip = TripKey['Value'];
-      final String line = trip[0];
-      if (stopData.configuredBuses.contains(line)) {
-        final String destination = trip[1];
-        String timeString = trip[2];
-        if (timeString.substring(timeString.length - 1) == '*') {
-          timeString = timeString.substring(0, timeString.length - 1);
-        }
-        final int timeRemaining = int.parse(timeString);
-        final Trip newTrip = Trip(
-            line: line, destination: destination, timeRemaining: timeRemaining);
-        tripList.add(newTrip);
+    final configuredBuses = stopData.configuredBuses;
+
+    final tripList = List<Trip>();
+
+    for (var entry in tableEntries) {
+      final rawBusInformation = entry.querySelectorAll('td');
+
+      final busLine = rawBusInformation[0].querySelector('ul > li').text.trim();
+
+      if(!configuredBuses.contains(busLine)){
+        continue;
       }
+
+      final busDestination = rawBusInformation[0].text
+          .replaceAll('\n', '')
+          .replaceAll('\t', '')
+          .replaceAll(' ', '')
+          .replaceAll('-', '')
+          .substring(busLine.length + 1);
+
+      final busTimeRemaining = getBusTimeRemaining(rawBusInformation);
+
+      final Trip newTrip = Trip(
+          line: busLine,
+          destination: busDestination,
+          timeRemaining: int.parse(busTimeRemaining)
+      );
+
+      tripList.add(newTrip);
     }
 
-    tripList.sort((a, b) => a.compare(b));
     return tripList;
   }
 
+  static String getBusTimeRemaining(rawBusInformation){
+    final busTimeOfArrival = rawBusInformation[1].text.trim();
+    var busTimeRemaining = '0';
+
+    if(busTimeOfArrival != 'a passar'){
+      busTimeRemaining = rawBusInformation[2].text;
+      if(busTimeRemaining.contains('min')){
+        busTimeRemaining =
+            busTimeRemaining.substring(0, busTimeRemaining.length - 3);
+      }
```

Comment:
Maybe it would be simpler to extract the numerical data from the string using a regex, thus removing the need for the substring check, just in case they change things

Reply:
Yes that's a great idea. Thank you!

Reply:
About the merge with #279 , they are both about the busstop feature, but in different ways, they work without each other, and since they resolve different issues, I think its indifferent to merge it to develop or to this branch first, either way, rebases will have to be made in both. @andrefmrocha, what do you think?

Reply:
(should this be under this thread?)
I think they should be looked into as separate things and wtv is merged first is fine.

Reply:
(I can't respond to the review comment) Fine by me.